proto/ca_service.proto

```diff
@@ -0,0 +1,63 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package proto;
+
+// Service definition of IstioCAService that can dynamically verify the CSR for
+// service identities for Istio services running on various platforms and
+// approve or disapprove the request after verifying the Node Agent credentials
+// provided as a part of the Certificte Signing Request.
+service IstioClusterCAService {
+
```

Comment:
When we are talking about Istio on hybrid clusters what is the composition of each cluster in the hybrid cluster? I don't think cluster is over-used. I think it is used more to describe a set of nodes that can work together in isolation. And it is used so much because it is better understood in the community.  Note that the service definition is only for the non-k8s cluster cases. In this case, we have an Istio cluster that will have its own CA, its own Pilot an ingress and egress proxy.